app/src/main/java/com/x8bit/bitwarden/ui/platform/components/toggle/BitwardenUnlockWithBiometricsSwitch.kt

```diff
@@ -17,16 +18,31 @@ import com.x8bit.bitwarden.R
  */
 @Composable
 fun BitwardenUnlockWithBiometricsSwitch(
-    isBiometricsSupported: Boolean,
+    biometricSupportStatus: BiometricSupportStatus,
     isChecked: Boolean,
     onDisableBiometrics: () -> Unit,
     onEnableBiometrics: () -> Unit,
     modifier: Modifier = Modifier,
 ) {
-    if (!isBiometricsSupported) return
+    if (biometricSupportStatus == BiometricSupportStatus.NOT_SUPPORTED) return
+    val biometricsDescription: String = when (biometricSupportStatus) {
+        BiometricSupportStatus.CLASS_3_SUPPORTED -> {
+            stringResource(R.string.class_3_biometrics_description)
+        }
+        BiometricSupportStatus.CLASS_2_SUPPORTED -> {
+            stringResource(R.string.class_2_biometrics_description)
+        }
+
+        BiometricSupportStatus.NOT_SUPPORTED -> error(
+            "Should not be called when BiometricSupportStatus is NOT_SUPPORTED",
+        )
```

Comment:
🎨 Can we `return` here instead of throwing? That would make the pre-check on L27 unnecessary.